### PR TITLE
[folly] update to 2023.05.08.00

### LINKED ISSUE #28475: [folly] update to <2022.12.19.00>
Library name: folly

New version number: 2022.12.19.00

https://github.com/facebook/folly/releases/tag/v2022.12.19.00

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fix #28475 

Update folly to the latest version 2023.05.08.00

Features `bzip2,libsodium,lz4,lzma,snappy,zlib,zstdare` tested successfully in the following triplet:

- x64-windows
- x64-windows-static

### CODE REVIEW

File: ports/folly/portfile.cmake [hunk 1]
```diff
@@ -8,8 +8,8 @@ vcpkg_add_to_path("${PYTHON3_DIR}")
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO facebook/folly
-    REF d8ed9cd2869c74b00fa6f1a7603301183f5c2249 #v2022.10.31.00
-    SHA512 55040dadb8a847f0d04c37a2dce920bb456a59decebc90920831998df9671feb33daf1f4235115adcce5eb9c469b97b9d96fa7a67a5914c434ebc1efc04f4770
+    REF 4c1964ddcecd157b478166c4c360fe95ad06c042 #v2022.10.31.00
```
Comment (kobykahane) [inline ports/folly/portfile.cmake:11]:
```suggestion
    REF 4c1964ddcecd157b478166c4c360fe95ad06c042 #v2022.12.19.00
```

**Comment by MonicaLiu0311** [comment]
The latest version: https://github.com/microsoft/vcpkg/issues/28745.

**Comment by YunchengLiu** [comment]
Will this PR continue?

**Comment by jiayuehua** [comment]
#31657 , has updated to 2023.05.15

**Comment by JonLiu1993** [comment]
Duplicate of #31657.

**Pull request closed without merging**